packages/devtools/src/formats/runtime.ts

```diff
@@ -249,6 +251,35 @@ export class RuntimeModelBuilder {
   }
 }
 
+function applyIndexing(
```

Comment:
this function is mutating its inputs which is a bit non-obvious if you don't read carefully.  Can you please add some doccomments explaining what this function is doing?